x-pack/filebeat/input/awss3/s3.go

```diff
@@ -16,16 +16,16 @@ import (
 	"github.com/elastic/beats/v7/libbeat/beat"
 )
 
-func createS3API(ctx context.Context, config config, awsConfig awssdk.Config) (*awsS3API, error) {
-	s3Client := s3.NewFromConfig(awsConfig, config.s3ConfigModifier)
-	regionName, err := getRegionForBucket(ctx, s3Client, config.getBucketName())
+func (in *s3PollerInput) createS3API(ctx context.Context) (*awsS3API, error) {
+	s3Client := s3.NewFromConfig(in.awsConfig, in.config.s3ConfigModifier)
+	regionName, err := getRegionForBucket(ctx, s3Client, in.config.getBucketName())
 	if err != nil {
 		return nil, fmt.Errorf("failed to get AWS region for bucket: %w", err)
 	}
 	// Can this really happen?
-	if regionName != awsConfig.Region {
-		awsConfig.Region = regionName
-		s3Client = s3.NewFromConfig(awsConfig, config.s3ConfigModifier)
+	if regionName != in.awsConfig.Region {
+		in.awsConfig.Region = regionName
+		s3Client = s3.NewFromConfig(in.awsConfig, in.config.s3ConfigModifier)
```

Comment:
➕ for using pointers. From first look, old code should have worked without any issue. But seems, there's some buggy optimization from Go, making the call at 28 to have same aws config copy as used in line 20 😁 

Good work catching this and fixing 🙌